### PR TITLE
Add support for CSS source maps

### PR DESCRIPTION
**Pull request by johanblumenberg**
Add support for generating css sourcemaps using gulp-sourcemaps.

https://www.npmjs.com/package/gulp-sourcemaps

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -24,7 +26,26 @@ module.exports = function () {
 		try {
 			var ret = rework(file.contents.toString(), {source: file.path});
 			plugins.forEach(ret.use.bind(ret));
-			file.contents = new Buffer(ret.toString(options));
+
+			if (file.sourceMap) {
+				options.sourcemap = true;
```
Comment (sindresorhus) [inline index.js]: You need to run `options` through `Object.assign` on line 12, so you don't modify the user's object.

**Review by sindresorhus — CHANGES_REQUESTED** [review]

File: index.js (hunk 1 above)
Comment (johanblumenberg) [reply index.js]: Fixed

**Comment by johanblumenberg** [comment]
I changed to keep the original options object untouched, and I also added a couple of test cases.

**Comment by sindresorhus** [comment]
You need to fix the lint issues: https://travis-ci.org/sindresorhus/gulp-rework/jobs/310684900 Running `npx xo --fix` (with latest npm installed) should take care of most of it automatically.

**Comment by johanblumenberg** [comment]
@sindresorhus I rebased on latest and fixed the issues

**Comment by johanblumenberg-symphony** [comment]
@sindresorhus What's blocking this PR?

**Pull request merged**